Kernel/Process.cpp

```diff
@@ -2504,13 +2504,33 @@ KResultOr<siginfo_t> Process::do_waitid(idtype_t idtype, int id, int options)
         auto* waitee_thread = Thread::from_tid(waitee_pid);
         if (!waitee_thread)
             return KResult(-ECHILD);
-        ASSERT(waitee_thread->state() == Thread::State::Stopped);
+        ASSERT((options & WNOHANG) || waitee_thread->state() == Thread::State::Stopped);
         siginfo_t siginfo;
         memset(&siginfo, 0, sizeof(siginfo));
         siginfo.si_signo = SIGCHLD;
         siginfo.si_pid = waitee_process->pid();
         siginfo.si_uid = waitee_process->uid();
-        siginfo.si_code = CLD_STOPPED;
+
+        // FIXME: I _really_ don't know what this should actually do
```

Comment:
POSIX @ https://pubs.opengroup.org/onlinepubs/9699919799/functions/waitid.html
>The application shall ensure that the infop argument points to a siginfo_t structure. If waitid() returns because a child process was found that satisfied the conditions indicated by the arguments idtype and options, then the structure pointed to by infop shall be filled in by the system with the status of the process; the si_signo member shall be set equal to SIGCHLD. If waitid() returns because WNOHANG was specified and status is not available for any process specified by idtype and id, then the si_signo and si_pid members of the structure pointed to by infop shall be set to zero and the values of other members of the structure are unspecified.

Linux @ http://man7.org/linux/man-pages/man2/sigaction.2.html
>The si_code field inside the siginfo_t argument that is passed to a SA_SIGINFO signal handler is a value (not a bit mask) indicating why this signal was sent.  ...
>For a non-ptrace(2) event, the values that can appear in si_code are described in the remainder of this section.

So POSIX doesn't go into any detail about `si_code`, really. I think maybe we can skip this field until we know we actually need it for communicating something. No need to bend over backwards just to imitate some obscure part of Linux kernel behavior.

Reply:
Well that bit is trying to communicate whether the child is running, I guess it's fine to set it to whatever since POSIX doesn't really specify?

Reply:
I guess we can leave it like it until it breaks something or does something unexpected?
it seems to work fine for its current use-case

Reply:
Either way is fine really. :)

Reply:
Then let's just go with this
> too lazy to change it anway